shards/eval.go

```diff
@@ -39,6 +41,44 @@ func (s *typeRepoSearcher) Search(ctx context.Context, q query.Q, opts *zoekt.Se
 	return s.Searcher.Search(ctx, q, opts)
 }
 
+func (s *typeRepoSearcher) StreamSearch(ctx context.Context, q query.Q, opts *zoekt.SearchOptions, sender stream.Sender) (err error) {
+	tr, ctx := trace.New(ctx, "typeRepoSearcher.StreamSearch", "")
+
+	a := struct {
+		sync.Mutex
+		n     int
+		stats *zoekt.Stats
+	}{
+		stats: new(zoekt.Stats),
+	}
+
+	tr.LazyLog(q, true)
+	tr.LazyPrintf("opts: %+v", opts)
+	defer func() {
+		tr.LazyPrintf("num files: %d", a.n)
+		tr.LazyPrintf("stats: %+v", a.stats)
+		if err != nil {
+			tr.LazyPrintf("error: %v", err)
+			tr.SetError(err)
+		}
+		tr.Finish()
+	}()
```

Comment:
these stats don't seem _that_ useful to log? I would simplify and just directly pass in sender?